# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/AntiTrackingUtils.h
# Commit: 518cc5289f58
# Full Hash: 518cc5289f58447ec9ab4d8373c749c15dc9fb90
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2021-10-28 03:37:11
# Regressor Bug: 1725996
# File Overlap Count: 1
# Description:
#   Bug 1725996 - Part 2: Implement AntiTrackingUtils::IsThirdPartyContext(). r=dimi
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D127627
# ==============================================================================

diff -r f41b0cceba4e -r 518cc5289f58 toolkit/components/antitracking/AntiTrackingUtils.h
--- a/toolkit/components/antitracking/AntiTrackingUtils.h	Wed Oct 27 22:56:51 2021 +0000
+++ b/toolkit/components/antitracking/AntiTrackingUtils.h	Wed Oct 27 22:56:51 2021 +0000
@@ -121,6 +121,10 @@
   // Otherwise, it will continue to check if the window is third-party.
   static bool IsThirdPartyWindow(nsPIDOMWindowInner* aWindow, nsIURI* aURI);
 
+  // Given a browsing context, this function determines if this browsing context
+  // is considered as a third party in respect to the top-level context.
+  static bool IsThirdPartyContext(dom::BrowsingContext* aBrowsingContext);
+
   static nsCString GrantedReasonToString(
       ContentBlockingNotifier::StorageAccessPermissionGrantedReason aReason);
 
